x-pack/platform/plugins/shared/streams_app/public/components/data_management/stream_detail_management/advanced_view/settings.tsx

```diff
@@ -79,10 +84,11 @@ export function Settings({
     [definition]
   );
   const [settings, setSettings] = useState<Record<string, Setting>>({});
-  const hasChanges = useMemo(
-    () => !isEqual(originalSettings, settings),
-    [originalSettings, settings]
-  );
+  const hasChanges = useMemo(() => {
+    const keys = [...new Set([...Object.keys(originalSettings), ...Object.keys(settings)])];
+    return keys.some((key) => originalSettings[key]?.value !== settings[key]?.value);
```

Comment:
There is a small logic issue here. 

On first render the `settings` in the state is an empty object. 

If:
`originalSettings['index.number_of_shards']?.value !== settings['index.number_of_shards']?.value` that would lead to: "1" !== undefined → true

So basically on first render, hasChanges would be true, potentially leading to a flash of the bar. In practice we might not see it, but it might be best to be on the safe side.

My proposal would be to initiate the state from the start:

```
const [settings, setSettings] = useState<Record<string, Setting>>(() => 
  toStringValues(definition.stream.ingest.settings, definition.effective_settings)
);
```